ui/app/AppLayouts/Wallet/SendModal.qml

```diff
@@ -1,27 +1,188 @@
 import QtQuick 2.13
 import QtQuick.Controls 2.13
 import QtQuick.Layouts 1.13
+import QtQuick.Dialogs 1.3
 import "../../../imports"
 import "../../../shared"
 import "./components"
 
 ModalPopup {
-    id: popup
+    id: root
 
     //% "Send"
     title: qsTrId("command-button-send")
     height: 700
 
-    onOpened: {
-        sendModalContent.amountInput.selectedAmount = ""
-        sendModalContent.passwordInput.text = ""
-        sendModalContent.amountInput.forceActiveFocus(Qt.MouseFocusReason)
+    property MessageDialog sendingError: MessageDialog {
+        id: sendingError
+        title: qsTr("Error sending the transaction")
+        icon: StandardIcon.Critical
+        standardButtons: StandardButton.Ok
     }
+    property MessageDialog sendingSuccess: MessageDialog {
+        id: sendingSuccess
+        //% "Success sending the transaction"
+        title: qsTrId("success-sending-the-transaction")
+        icon: StandardIcon.NoIcon
+        standardButtons: StandardButton.Ok
+        onAccepted: {
+            root.close()
+        }
+    }
+
+    onClosed: {
+        stack.reset()
+    }
+
+    function sendTransaction() {
+        let responseStr = walletModel.sendTransaction(selectFromAccount.selectedAccount.address,
+                                                 selectRecipient.selectedRecipient.address,
+                                                 txtAmount.selectedAsset.address,
+                                                 txtAmount.selectedAmount,
+                                                 gasSelector.selectedGasLimit,
+                                                 gasSelector.selectedGasPrice,
+                                                 transactionSigner.enteredPassword)
+        let response = JSON.parse(responseStr)
 
-    SendModalContent {
-        id: sendModalContent
-        closePopup: function () {
-            popup.close()
+        if (response.error) {
+            // It's an error
+            if (response.error.includes("could not decrypt key with given password")){
+                transactionSigner.validationError = qsTr("Wrong password")
+                return
+            }
+            sendingError.text = response.error
+            return sendingError.open()
+        }
+
+        sendingSuccess.text = qsTr("Transaction sent to the blockchain. You can watch the progress on Etherscan: %2/%1").arg(response.result).arg(walletModel.etherscanLink)
+        sendingSuccess.open()
+    }
+
+    TransactionStackView {
+        id: stack
+        height: parent.height
+        anchors.fill: parent
+        anchors.leftMargin: Style.current.padding
+        anchors.rightMargin: Style.current.padding
+        onGroupActivated: {
+            root.title = group.headerText
+            btnNext.label = group.footerText
+        }
+        TransactionStackGroup {
```

Comment:
First off, great work on the `TransactionStackView` + `TransactionStackGroup` composition.
I've got a few thoughts here, curious what you think:

- `TransactionStackView` really is a `StackView` with some custom cherries on top. Would be cool if we could try leveraging native `StackView` instead of making it an `Item`. We might not even need a custom StackView here at all and could implement it inline. Seems like the only thing it adds is `reset()` which could probably be added to `StackView` handlers.
- `TransactionStackGroup` really is just a view that holds a header and a footer and happens to have some knowledge about validation and resetting (which is expected by `TransactionStackView`. This "pattern" of having a dedicated `validate()` method plus its `isValid` state attached to form controls is quite nice. I think however it is so generic, that it probably makes sense to have something like a `FormGroup` component that can have #n Form controls (which also come with their own validity state. The FormGroup is then used in different component views that are pushed on to the StackView.

So all in all, I think it'd be nice if we split `TransactionStackGroup` into "SomethingSomethingVIew" with dedicated `StackView` handlers etc. and then introduce a FormGroup component that knows how to validate its form controls.

Just thinking out loud here.. also not saying this needs to happen as part of this PR.

Wdyt?

Reply:
Please, don't make this PR bigger lol 🙏

Reply:
Added a `FormGroup` component as suggested, changed `TransactionStackGroup` to `TransactionFormGroup` which now looks like:
```
FormGroup {
    id: root
    property string headerText
    property string footerText
}
```
Soz jono.